services/gateway_manager/src/main/java/com/futurewei/alcor/gatewaymanager/service/GatewayService.java

```diff
@@ -0,0 +1,188 @@
+package com.futurewei.alcor.gatewaymanager.service;
+
+import com.futurewei.alcor.common.db.CacheException;
+import com.futurewei.alcor.common.executor.AsyncExecutor;
+import com.futurewei.alcor.gatewaymanager.config.ExceptionMsgConfig;
+import com.futurewei.alcor.gatewaymanager.dao.GWAttachmentRepository;
+import com.futurewei.alcor.gatewaymanager.dao.GatewayRepository;
+import com.futurewei.alcor.gatewaymanager.entity.GWAttachment;
+import com.futurewei.alcor.gatewaymanager.entity.ResourceType;
+import com.futurewei.alcor.gatewaymanager.entity.StatusEnum;
+import com.futurewei.alcor.web.entity.gateway.*;
+import com.futurewei.alcor.web.restclient.GatewayManagerRestClinet;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.function.Supplier;
+
+@Slf4j
+@Service
+@ComponentScan("com.futurewei.alcor.web.restclient")
+public class GatewayService {
+
+    @Autowired
+    private GatewayRepository gatewayRepository;
+
+    @Autowired
+    private GWAttachmentRepository gwAttachmentRepository;
+
+    @Autowired
+    private GatewayManagerRestClinet restClinet;
+
+    private static final String GATEWAY_DESCRIPTION = "internal gateway";
+    private static final String GATEWAY_SUFFIX_NAME = " name";
+    private static final String ATTACHMENT_NAME_PREFIX = "attachment - ";
+
+
+    public GatewayService() {
+    }
+
+    public GatewayInfo createGatewayInfo(String projectId, VpcInfo vpcInfo) throws Exception {
+
+        //TODO check whether the project type is zeta by projectId
+
+        // create GatewayInfo entity in the DB
```

Comment:
@songxiaoyan Please re-phrase the comment here. We are not creating a GatewayInfo enity in GM's DB.